tools/odbc/result.cpp

```diff
@@ -92,11 +252,336 @@ SQLRETURN SQLGetData(SQLHSTMT statement_handle, SQLUSMALLINT col_or_param_num, S
 			}
 			return SQL_SUCCESS;
 		}
+		case SQL_C_TYPE_DATE: {
+			date_t date;
+			switch (val.type().id()) {
+			case LogicalTypeId::DATE:
+				date = val.GetValue<date_t>();
+				break;
+			case LogicalTypeId::TIMESTAMP:
+			case LogicalTypeId::TIMESTAMP_SEC:
+			case LogicalTypeId::TIMESTAMP_MS:
+			case LogicalTypeId::TIMESTAMP_NS: {
+				try {
+					date = duckdb::CastTimestampToDate::Operation<timestamp_t, date_t>(val.GetValue<timestamp_t>());
+				} catch (duckdb::Exception &ex) {
+					stmt->error_messages.emplace_back(ex.what());
+					return SQL_ERROR;
+				}
+				break;
+			}
+			case LogicalTypeId::VARCHAR: {
+				string val_str = val.GetValue<string>();
+				try {
+					date = duckdb::CastToDate::Operation<string_t, date_t>(string_t(val_str));
+				} catch (duckdb::Exception &ex) {
+					stmt->error_messages.emplace_back(ex.what());
+					return SQL_ERROR;
+				}
+				break;
+			}
+			case LogicalTypeId::HUGEINT: {
+				hugeint_t val_hint = val.GetValue<hugeint_t>();
+				if (!duckdb::TryCast::Operation<hugeint_t, date_t>(val_hint, date)) {
```

Comment:
Hugeint -> Date conversion seems like it lacks a semantic meaning. Hugeint is only used to represent numbers, so what date is the number "42"? I think this conversion should be removed and it should throw an error here.